File: kinotic-js/workspace/packages/core/src/api/reconcile/Reconcilable.ts
import type { Watched } from './Watched'
import type { ReconcileState } from './ReconcileState'

/**
 * A watched record that also carries intent: what it should be, beside what its authority last
 * reported it is, so a worker can converge the world to it whenever the two differ.
 */
export interface Reconcilable<S> extends Watched {

    state: ReconcileState<S>
}

File: kinotic-domain/src/main/java/org/kinotic/domain/api/reconcile/Reconcilable.java
package org.kinotic.domain.api.reconcile;

/**
 * A {@link Watched} record that also carries intent: what it should be, beside what its authority
 * last reported it is, so a worker can converge the world to it whenever the two differ.
 *
 * @param <S> the record's state, the same shape on both sides
 */
public interface Reconcilable<S> extends Watched {

    @Override
    ReconcileState<S> getState();
}
